feat(replays): change replays processor to consume directly from relay event

JoshFerge opened the pull request:
With some simplifications to replay's architecture, we can directly consume replay events that are emitted from relay. This PR changes the processor to consume this schema, and also removes some unnecessary logic around promoted tags.

Replays is not live yet, so there is no concern about worrying about the old schema.

Updated architecture page here: https://www.notion.so/sentry/Replay-Backend-Architecture-Update-7-19-7c4ecc8af3fb4027992f3ee75dae9529

evanh commented on snuba/datasets/replays_processor.py:
```diff
@@ -27,146 +27,121 @@
+    ) -> None:
+        processed["replay_id"] = str(uuid.UUID(replay_event["replay_id"]))
+        processed["sequence_id"] = replay_event["sequence_id"]
+        processed["trace_ids"] = replay_event.get("trace_ids")
```
`trace_ids` is not a nullable column from what I can see in the schema. I'm not sure it will work if you try to write `None` to Clickhouse.

JoshFerge replied:
👍🏼

evanh commented on snuba/datasets/replays_processor.py:
```diff
@@ -27,146 +27,121 @@
-        processed["trace_ids"] = event_dict["trace_ids"]
-
+    def _add_user_field_tag(
+        self, processed: Any, replay_event: ReplayEventDict
```
```suggestion
        self, processed: MutableMapping[str, Any], replay_event: ReplayEventDict
```

evanh commented on snuba/datasets/replays_processor.py:
```diff
@@ -27,146 +27,121 @@
-
-        tags: Mapping[str, Any] = _as_dict_safe(event_dict["data"].get("tags", None))
+        tags: MutableMapping[str, Any] = _as_dict_safe(replay_event.get("tags", None))
+        processed["title"] = tags.get("transaction")
```
`title` is not nullable.

JoshFerge replied:
👍🏼 let me know if it's preferred to write empty string, or we can also make it nullable instead. i saw this: https://stackoverflow.com/questions/63057886/clickhouse-string-field-disk-usage-null-vs-empty

evanh commented on snuba/datasets/replays_processor.py:
```diff
@@ -27,146 +27,121 @@
+        self, processed: Any, replay_event: ReplayEventDict
+    ) -> None:
+        if "user" not in replay_event:
+            return
```
`user` and `user_hash` are not nullable in your schema.

JoshFerge replied:
👍🏼

evanh reviewed (commented):
I think you need to double check this processor against your schema for which columns are nullable. There are more than just the ones I commented on.

JoshFerge commented:
@evanh i've adjusted the processor to handle the null cases you've mentioned (and other possible ones), and that the clickhouse writes work. i've adjusted the schema a bit, in #2972, which is required for this PR. I've seen that if a string is omitted but not nullable (e.g. title), the writes are still successful, and i believe an empty string will exist in that column.

evanh reviewed (approved):
Thanks for that, looks good.
